### ISSUE TITLE
Style.css not compiling on development mode

### ISSUE THREAD

**Issue by codexpert** [apvarun/ng-tailwindcss-jit#1]
Hello Varun,

Really appreciate your work. I'm having the same problem, style file not updating on module html change. 

Here is my start command `"start": "TAILWIND_MODE='watch' node scripts/com.ts && ng serve",`

Any workaround?

**Comment by apvarun** [comment]
Hey, few questions

1. Did you try running the script? `npx ng-tailwindcss-jit`
2. Have you added html files to the list of purge files in tailwindcss config?

Here is a working repo for demo. Maybe you can refer this: https://github.com/apvarun/demo-ng-tailwindcss-jit

**Comment by codexpert** [comment]
Yeah I did and no luck :( 

Here is the tailwind config file

```
module.exports = {
  prefix: '',
  purge: ["./src/**/*.{html,scss}"],
  darkMode: 'media', // or 'media' or 'class'
  theme: {
    extend: {},
  },
  variants: {
    extend: {},
  },
  plugins: [
    require('@tailwindcss/aspect-ratio'), 
    require('@tailwindcss/forms'), 
    require('@tailwindcss/line-clamp'), 
    require('@tailwindcss/typography')
  ],
};
```
Compared our setup with your demo and i'm sure i'm not missing anything but its not working :(

**Comment by apvarun** [comment]
That weird. Can you try running the `postinstall` script manually?

If that also didn't work, maybe a repo that you can share would be great for me to have a look.

**Comment by RRakib** [comment]
Hi Varun,

Actually the problem is connected to the environment variable **TAILWIND_MODE**. And i think this is Mac specific. I have come up with a solution. 

Create a bash file and put it on the root folder.

#### **tailwindpatch.sh**

    #!/bin/sh

    if [[ "$OSTYPE" == "darwin"* ]]
      then
        echo "# Setting up environment variable"
        export TAILWIND_MODE='watch'

        echo "# Updating tailwindcss package path"
        node ./scripts/ng-tailwindcss-jit.js
    fi


    echo "# Starting the server"
    yarn start

And finally add this script in _package.json_

#### **package.json**

    "patch": "chmod +x ./tailwindpatch.sh && ./tailwindpatch.sh",

All set. Now just run
    
    yarn patch

You might need root privilege to make the script executable. In that case just run the above command as root

**Comment by apvarun** [comment]
@RRakib Yes, this could be an issue, but aren't you doing the same thing essentially? I don't see an else case in the patch

**Comment by codexpert** [comment]
@apvarun After applying @RRakib patch it working as expected. Not sure why ENV not working on M1 Mac.

**Comment by RRakib** [comment]
@apvarun I am checking the os with if statement. That's why there are no else case. I think setting up the variable through package.json doesn't work on mac. So i am directly injecting it to the environment variable. Also it is session dependent.

**Comment by apvarun** [comment]
> package.json doesn't work on mac

This is a bit weird. I'm on mac and it does work. @RRakib Will you be able to try using cross-env for this instead of script? If that works, we can use that to make it work for other OS's as well

**Comment by RRakib** [comment]
> pkg.scripts.start = `TAILWIND_MODE='watch' ${pkg.scripts.start}`;

@apvarun Found the issue. As my script for `yarn start` was something like `TAILWIND_MODE='watch' ......&&.......&& ng serve`. It was not accepting the environment variable. So after i manually put the variable in front of ng serve like  `......&&.......&& TAILWIND_MODE='watch' ng serve` and now it works.

**Comment by codexpert** [comment]
@RRakib You are life saver 👯‍♂️

**Comment by apvarun** [comment]
Thanks @RRakib  and @codexpert for helping with finding the issue. I've updated the package to use cross-env and also fixed the issue you mentioned earlier in #2.
